include/ceed/ceed.h

```diff
@@ -140,14 +140,20 @@ typedef int32_t CeedInt;
 typedef ptrdiff_t CeedSize;
 
 /// Scalar (floating point) types
-///
 /// @ingroup Ceed
 typedef enum {
   /// Single precision
   CEED_SCALAR_FP32,
   /// Double precision
-  CEED_SCALAR_FP64
+  CEED_SCALAR_FP64,
+  /// Total number of allowed scalar precision types
+  CEED_NUM_PRECISIONS,
 } CeedScalarType;
+/// Struct for holding data in multiple precisions for mixed-precision-enabled
+/// backends
+typedef struct {
+  void *values[CEED_NUM_PRECISIONS]; // Size equals CEED_NUM_PRECISIONS
+} CeedScalarArray;
 /// Base scalar type for the library to use: change which header is 
 /// included to change the precision.
 #include "ceed-f64.h"
```

Comment:
This would allow a user to include the type-specific header before `ceed.h`, and thus be able to select the default at file granularity.
```suggestion
#ifndef CEED_SCALAR_TYPE
#  include "ceed-f64.h"
#endif
```

Do you think we can avoid referencing `CEED_SCALAR_TYPE` in library code? I think that's the eventual goal.